web-marketplace/src/pages/BuyCredits/hooks/usePurchase.ts

```diff
@@ -213,42 +268,65 @@ export const usePurchase = () => {
               );
             },
             onSuccess: async (deliverTxResponse?: DeliverTxResponse) => {
-              setProcessingModalAtom(atom => void (atom.open = false));
+              setTxHash(deliverTxResponse?.transactionHash);
 
-              // TODO https://regennetwork.atlassian.net/browse/APP-361
-              // We need to display a custom success modal here
-              // instead of the regular one
-              setTxSuccessfulModalAtom(atom => {
-                atom.open = true;
-                // atom.cardItems = cardItems; // TODO
-                // atom.title = _(POST_CREATED);
-                atom.buttonTitle = _(VIEW_CERTIFICATE);
-                // atom.buttonLink = buttonLink;
-                atom.txHash = deliverTxResponse?.transactionHash;
-              });
+              // In case of retirement, it's handled in useFetchRetirementForPurchase
+              if (!retiring) {
+                setProcessingModalAtom(atom => void (atom.open = false));
+                setTxBuySuccessfulModalAtom(atom => {
+                  atom.open = true;
+                  atom.cardItems = getCardItems({
+                    retiring,
+                    creditsAmount,
+                    currencyAmount,
+                    project,
+                    currency,
+                    displayDenom,
+                  });
+                  atom.buttonTitle = _(VIEW_PORTFOLIO);
+                  atom.onButtonClick = () =>
+                    setTxBuySuccessfulModalAtom(
+                      atom => void (atom.open = false),
+                    );
+                  atom.txHash = deliverTxResponse?.transactionHash;
+                  atom.steps = getSteps(paymentOption, retiring);
+                  atom.description = email
+                    ? `${_(EMAIL_RECEIPT)} ${email}`
+                    : undefined;
+                });
 
-              // Reload sell orders
-              await reactQueryClient.invalidateQueries({
-                queryKey: [SELL_ORDERS_EXTENTED_KEY],
-              });
+                await reactQueryClient.invalidateQueries({
+                  queryKey: [SELL_ORDERS_EXTENTED_KEY],
+                });
 
-              // Reset BuyCredits forms
-              handleSuccess();
+                // Reset BuyCredits forms
+                handleSuccess();
+                navigate(`/profile/portfolio`);
```

Comment:
TODO update to use dashboard when https://github.com/regen-network/regen-web/pull/2505 merged